features/functests/performance/performance.go

```diff
@@ -50,20 +63,145 @@ var _ = Describe("performance", func() {
 	var _ = Context("Pre boot tuning setup", func() {
 		It("Should contain a custome initrd image in boot loader", func() {
 			nodes, err := clients.K8s.CoreV1().Nodes().List(metav1.ListOptions{
-				LabelSelector: "node-role.kubernetes.io/worker-rt=",
+				LabelSelector: perfWorkerNodesLabel,
+			})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(nodes.Items)).Should(BeNumerically(">", 0), "cannot find nodes labeled as "+perfWorkerNodesLabel)
+
+			for _, node := range nodes.Items {
+				mcd, err := mcdForNode(&node)
+				Expect(err).ToNot(HaveOccurred())
+				mcdName := mcd.ObjectMeta.Name
+				By("executing the command \"grep -R  initrd /rootfs/boot/loader/entries/\" inside the pod " + mcdName)
+				bootLoaderEntries, err := exec.Command("oc", "rsh", "-n", mcd.ObjectMeta.Namespace, "-c" ,
+					perfMachineConfigDaemonContainer, mcdName, "grep", "-R", "initrd", "/rootfs/boot/loader/entries/").CombinedOutput()
+				Expect(err).ToNot(HaveOccurred())
+				Expect(strings.Contains(string(bootLoaderEntries), "iso_initrd.img")).To(BeTrue(), "cannot find iso_initrd.img entry among the bootloader entries")
+			}
+		})
+	})
+
+	var _ = Context("Network latency parameters", func() {
+		It("Should contain configuration injected through the 00-tuned-network-latency template", func() {
+			sysctlMap := map[string]string{
+				"net.core.busy_read":              "50",
+				"net.core.busy_poll":              "50",
+				"net.ipv4.tcp_fastopen":           "3",
+				"kernel.numa_balancing":           "0",
+				"kernel.sched_min_granularity_ns": "10000000",
+				"vm.dirty_ratio":                  "10",
+				"vm.dirty_background_ratio":       "3",
+				"vm.swappiness":                   "10",
+				"kernel.sched_migration_cost_ns":  "5000000",
+			}
+
+			nodes, err := clients.K8s.CoreV1().Nodes().List(metav1.ListOptions{
+				LabelSelector: perfWorkerNodesLabel,
+			})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(nodes.Items)).Should(BeNumerically(">", 0), "cannot find nodes labeled as "+perfWorkerNodesLabel)
+
+			for _, node := range nodes.Items {
+				mcd, err := mcdForNode(&node)
+				Expect(err).ToNot(HaveOccurred())
```

Comment:
can you return false here instead of Expect(err).ToNot ?

Reply:
no, not here, a few lines below in the `Eventually` block. There you need to return a string which does not fullfill the expectation, in order to rerun the oc command. `Expect(err).ToNot(HaveOccurred())` does fail the test and not retry.